src/components/NewActivityList.js

```diff
@@ -26,11 +26,15 @@ class NewActivityList extends Component {
   }
 
   componentDidMount() {
-    const userId = firebase.auth().currentUser.uid;
-    firebase.database().ref('/users/' + userId + '/settings').once('value').then(snapshot => {
-      const currentRate = (snapshot.val().currentRate * 0.4)
+    const { uid } = this.props.user
+    firebase.database().ref('/users/' + uid + '/settings').once('value').then(snapshot => {
+      if (snapshot.val().currentRate) {
```

Comment:
if snapshot.val() is null, does null.currentRate will break the app?
check what happens when you try to use key from null